upgrade/steps.go

```diff
@@ -1121,12 +1121,19 @@ var fetchLatestJavaGen = stepv2.Func03("Fetching latest Java Gen", func(ctx cont
 	return currentJavaGen, latestJavaGen.String(), true
 })
 
-var parseUpstreamProviderOrg = stepv2.Func11("Get UpstreamOrg from module version", func(ctx context.Context, upstreamMod module.Version) string {
+var parseUpstreamProviderOrg = stepv2.Func11E("Get UpstreamOrg from module version", func(ctx context.Context, upstreamMod module.Version) (string, error) {
 	// We expect tokens to be of the form:
 	//
 	//	github.com/${org}/${repo}/${path}
 	//
 	// The second chunk is the org name.
+
+	// Verify that the token is of valid format
+	goModRegexp := regexp.MustCompile("github.com/[a-zA-Z0-9-]*/[a-zA-Z0-9-]*")
```

Comment:
I know your following the helpful comment, but I'm pretty sure that the comment is wrong. 
```suggestion
	goModRegexp := regexp.MustCompile("[\.a-zA-Z0-9-]*/[a-zA-Z0-9-]*/[a-zA-Z0-9-]*")
```

Specifically, I think this will break `pulumi-gitlab`:

https://github.com/pulumi/pulumi-gitlab/blob/5e8f66b41c8250bcd117a1a45333db3b098bcddd/provider/go.mod#L11

Reply:
Indeed! thank you for catching that.